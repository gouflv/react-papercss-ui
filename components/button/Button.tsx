import classNames from 'classnames'
import React, { FC } from 'react'

export interface ButtonProps {
  type?: 'primary' | 'secondary' | 'success' | 'warning' | 'danger'
  size?: 'large' | 'small'
  plain?: boolean
  block?: boolean
  disabled?: boolean
  onClick?: () => void
}

const Button: FC<ButtonProps> = props => {
  const { type, size, plain, block, disabled, onClick } = props
  return (
    <button
      className={classNames('paper-btn', {
        [`btn-${type}${plain ? '-outline' : ''}`]: !!type,
        [`btn-${size}`]: !!size,
        'btn-block': block
      })}
      disabled={disabled}
      onClick={onClick}
    >
      {props.children}
    </button>
  )
}

export { Button }
