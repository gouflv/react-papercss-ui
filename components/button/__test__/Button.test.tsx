import React, { PropsWithChildren } from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import { Button, ButtonProps } from '../Button'

describe('Test Component', () => {
  let props: PropsWithChildren<ButtonProps>

  beforeEach(() => {
    props = {
      type: 'primary',
      children: 'Button Test'
    }
  })

  const renderComponent = () => render(<Button {...props} />)

  it('should have default className with default props', () => {
    const { container } = renderComponent()
    const cmp = screen.getByText('Button Test')
    expect(cmp).toHaveClass('paper-btn')
    expect(container).toMatchSnapshot()
  })

  it('should have secondary className with type set as secondary', () => {
    props.type = 'secondary'
    const { container } = renderComponent()
    const cmp = screen.getByText('Button Test')
    expect(cmp).toHaveClass('btn-secondary')
    expect(container).toMatchSnapshot()
  })

  it('should have sizing className with size set as large', () => {
    props.size = 'large'
    const { container } = renderComponent()
    const cmp = screen.getByText('Button Test')
    expect(cmp).toHaveClass('btn-large')
    expect(container).toMatchSnapshot()
  })

  it('should have block className with set block', () => {
    props.block = true
    const { container } = renderComponent()
    const cmp = screen.getByText('Button Test')
    expect(cmp).toHaveClass('btn-block')
    expect(container).toMatchSnapshot()
  })

  it('should have disabled attr with set disabled', () => {
    props.disabled = true
    const { container } = renderComponent()
    const cmp = screen.getByText('Button Test')
    expect(cmp).toHaveAttribute('disabled')
    expect(container).toMatchSnapshot()
  })

  it('should called event handler on click', () => {
    const onClick = (props.onClick = jest.fn())
    const { container } = renderComponent()
    fireEvent.click(screen.getByText('Button Test'))
    expect(onClick).toBeCalled()
    expect(container).toMatchSnapshot()
  })
})
