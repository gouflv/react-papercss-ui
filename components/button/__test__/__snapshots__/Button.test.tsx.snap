// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component should called event handler on click 1`] = `
<div>
  <button
    class="paper-btn btn-primary"
  >
    Button Test
  </button>
</div>
`;

exports[`Test Component should have block className with set block 1`] = `
<div>
  <button
    class="paper-btn btn-primary btn-block"
  >
    Button Test
  </button>
</div>
`;

exports[`Test Component should have default className with default props 1`] = `
<div>
  <button
    class="paper-btn btn-primary"
  >
    Button Test
  </button>
</div>
`;

exports[`Test Component should have disabled attr with set disabled 1`] = `
<div>
  <button
    class="paper-btn btn-primary"
    disabled=""
  >
    Button Test
  </button>
</div>
`;

exports[`Test Component should have secondary className with type set as secondary 1`] = `
<div>
  <button
    class="paper-btn btn-secondary"
  >
    Button Test
  </button>
</div>
`;

exports[`Test Component should have sizing className with size set as large 1`] = `
<div>
  <button
    class="paper-btn btn-primary btn-large"
  >
    Button Test
  </button>
</div>
`;
